internal/operations/operation_updater.go

```diff
@@ -93,6 +93,37 @@ func (ou *operationUpdater) pickWorker(ctx context.Context, id *fftypes.UUID, up
 	return ou.workQueues[worker]
 }
 
+// SubmitBulkOperationUpdates will wait for the commit to DB before calling the onCommit
+func (ou *operationUpdater) SubmitBulkOperationUpdates(ctx context.Context, updates []*core.OperationUpdate) error {
+	validUpdates := []*core.OperationUpdate{}
+	for _, update := range updates {
+		ns, _, err := core.ParseNamespacedOpID(ctx, update.NamespacedOpID)
+		if err != nil {
+			log.L(ctx).Warnf("Unable to update operation '%s' due to invalid ID: %s", update.NamespacedOpID, err)
+			continue
+		}
+
+		if ns != ou.manager.namespace {
```

Comment:
Would be good to think carefully if this is safe/needed. This API will only be implemented by updated connectors, so will be new environments (not migration ones).

If you can do the investigation to check into why this code was in the place it was copied from, I _think_ you'll find it's related to a migration scenario back from single-listener to multi-listener days.

If you can confirm that, then it is much safer to return an error to the connector code if it supplies bad data.